### PR TITLE
ta: pkcs11: Fix key generation for Ed25519

### PR DESCRIPTION
**Pull request by varder**
Fix the issues with the key generation in 

https://github.com/OP-TEE/optee_test/pull/618

### CODE REVIEW

File: core/tee/tee_svc_cryp.c [hunk 1]
```diff
@@ -476,7 +476,7 @@ const struct tee_cryp_obj_type_attrs tee_cryp_obj_ed25519_keypair_attrs[] = {
 
 	{
 	.attr_id = TEE_ATTR_ECC_CURVE,
-	.flags = TEE_TYPE_ATTR_SIZE_INDICATOR | TEE_TYPE_ATTR_GEN_KEY_REQ,
+	.flags = TEE_TYPE_ATTR_SIZE_INDICATOR | TEE_TYPE_ATTR_GEN_KEY_OPT,
```
Comment (jforissier) [inline core/tee/tee_svc_cryp.c:479]: Is it really what we want? See https://github.com/OP-TEE/optee_os/pull/5573. According to the GP spec (1.3.1) `TEE_ATTR_ECC_CURVE` is NOT a valid attribute for `TEE_TYPE_ED25519_KEYPAIR`.

Comment (varder) [reply core/tee/tee_svc_cryp.c]: Agree, there no need for the CURVE attribute.
Previously the key size was deduced based on curve when
in CMD_CREATE_OBJECT.
It is reworked to get key size from PKCS11_CKA_EC_POINT.
How do you think, is it Ok?
Also removed the code for ECC_CURVE attribute.

**Comment by jforissier** [comment]
Please squash the fixup, rebase onto master and properly explain what is wrong in commit 03e07432b that you are fixing.

**Comment by varder** [comment]
> Please squash the fixup, rebase onto master and properly explain what is wrong in commit [03e0743](https://github.com/OP-TEE/optee_os/commit/03e07432b68f9d78dbad4b9a8af50705948e7893) that you are fixing.

Fixed

File: ta/pkcs11/src/processing.c [hunk 2]
```diff
@@ -146,12 +146,19 @@ size_t get_object_key_bit_size(struct pkcs11_object *obj)
 
 		return a_size * 8;
 	case PKCS11_CKK_EC:
-	case PKCS11_CKK_EC_EDWARDS:
 		if (get_attribute_ptr(attrs, PKCS11_CKA_EC_PARAMS,
 				      &a_ptr, &a_size) || !a_ptr)
 			return 0;
 
 		return ec_params2tee_keysize(a_ptr, a_size);
+
```
Comment (klogg) [inline ta/pkcs11/src/processing.c:154]: Unnecessary blank lines around case block, here ...

File: ta/pkcs11/src/processing.c [hunk 3]
```diff
@@ -146,12 +146,19 @@ size_t get_object_key_bit_size(struct pkcs11_object *obj)
...
+	case PKCS11_CKK_EC_EDWARDS:
+		if (get_attribute_ptr(attrs, PKCS11_CKA_EC_POINT, NULL,
+				      &a_size))
+			return 0;
+
+		return a_size * 8;
+
```
Comment (klogg) [inline ta/pkcs11/src/processing.c:161]: ... and here

**Review by klogg — CHANGES_REQUESTED** [review]

File: ta/pkcs11/src/processing.c (hunk 2 above)
Comment (varder) [reply ta/pkcs11/src/processing.c]: fixed

File: ta/pkcs11/src/processing.c (hunk 3 above)
Comment (varder) [reply ta/pkcs11/src/processing.c]: fixed

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@linaro.org>`

**Pull request merged**